server/controllers/UserController.js

```diff
@@ -0,0 +1,74 @@
+import models from '../database/models';
+import helpers from '../helpers';
+
+const { FavoriteBook, Book } = models;
+const { responseMessage } = helpers;
+
+/**
+ * @class UserController
+ * @classdesc controller for all user functionality
+ */
+export default class UserController {
+  /**
+   * @description check if book is in database
+   * @param {object} request express request object
+   * @returns {object} book object
+   * @memberof UserController
+   */
+  static async findBook(request) {
+    const { id } = request.userData;
+    const { bookId } = request.params;
+    const favoritebook = await FavoriteBook.findOne({ where: { userId: id, bookId } });
+    const book = await Book.findOne({ where: { id: bookId } });
+    return [favoritebook, book, { userId: id, bookId: parseInt(bookId, 10) }];
+  }
+
+  /**
+   * @description user can favorite a book
+   * @param {object} request express request object
+   * @param {object} response express response object
+   * @returns {json} json
+   * @memberof UserController
+   */
+  static async favoriteBook(request, response) {
+    try {
+      const [favoritebook, book, param] = await UserController.findBook(request);
+      if (!book) return responseMessage(response, 404, { message: 'book not found' });
+      if (favoritebook) return responseMessage(response, 400, { message: 'book already in your favourites' });
+      const favorite = await FavoriteBook.create(param);
+      if (favorite) {
+        return responseMessage(response, 200, {
+          status: 'success',
+          message: 'favorited successfully'
```

Comment:
can I get **data** information of the book successfully favorited

Reply:
done